dialoptions.go

```diff
@@ -401,10 +401,12 @@ func WithTimeout(d time.Duration) DialOption {
 // returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
 //
-// Note: As of Go 1.21, the standard library overrides the OS defaults for
-// TCP keepalive time and interval to 15s.
-// To retain OS defaults, use a net.Dialer with the KeepAlive field set to a
-// negative value.
+// Note: As of Go 1.21, the standard library overrides the OS defaults for TCP
```

Comment:
I agree this wording is confusing, and have been trying to think of something better...  Maybe:

`Note: All supported releases of Go (as of December 2023) override the OS defaults for...`

Or "as of the Go 1.21 release" or "up to and including at least Go 1.21" for the parenthetical?

Reply:
Done. Thanks.